os/kernel/debug/mem_leak_checker.c

```diff
@@ -414,3 +414,19 @@ int run_mem_leak_checker(int checker_pid, char *bin_name)
 
 	return OK;
 }
+
+void run_all_mem_leak_checker(int checker_pid, char *bin_name)
+{
+	printf("Kernel :\n");
+	run_mem_leak_checker(checker_pid, "kernel");
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	bin_addr_info_t *bin_addr_info = (bin_addr_info_t *)get_bin_addr_list();
+	int bin_idx;
+	for (bin_idx = 1; bin_idx <= CONFIG_NUM_APPS; bin_idx++) {
```

Comment:
it started from 1, and you execute run_mem_leak_checker with "kernel" as a 2nd parameter.
so I guess 0 is always kernel.
Then can't we modify loop to start bin_idx from 0?

Reply:
0 index is reserved for common binary. So we cant use it for kernel.

Reply:
One question, then it seems that it doesn't include kernel, because here it doesn't check kernel.
0 is for common binary, then it never check common binary? or is it unncessary?

Reply:
Yes, we will not check common binary. Common binary is code that will be used by apps. So any allocation done from app through common binary code will still come under the app's allocation. So, it's not required.

Reply:
Could you leave some code comment why we check it from index 1?

Reply:
I have added a comment for that in the file. I am attaching the code for reference here.

https://github.com/Samsung/TizenRT/blob/2b9459d794b105c766350da270aa4e5b9e1820d6/os/kernel/binary_manager/binary_manager_data.c#L64-L66